spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/converter/AppBootSchemaVersionConverter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2015-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.shell.converter;
+
+import org.springframework.cloud.dataflow.core.AppBootSchemaVersion;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+
+/**
+ * Converts strings to {@link AppBootSchemaVersion}
+ *
+ * @author Chris Bono
+ * @author Corneil du Plessis
+ */
+@Component
+public class AppBootSchemaVersionConverter implements Converter<String, AppBootSchemaVersion> {
```

Comment:
I know what you are thinking... however, the only way this is getting picked up by the shell is if it is in this location. It must be that the shell is doing a component scan or something shell specific to get its converters. You can remove this and watch the newly added tests go boom.